angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/weather.html

```diff
@@ -1,5 +1,21 @@
 <div>
-    <loading-gif data-object='weatherData'></loading-gif>
+
+    <div ng-if="!weather.location">
+		<div ng-model="noLocations" ng-init="noLocations=true">
+			<i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
+			<div class="warning-message-weather-widget">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
+		</div>
+		
+		<div ng-if="!noLocations"id="loading">
+    		<loading-gif data-object='weatherData'></loading-gif>
+    	</div>
+	</div>
+
+    <div ng-if="weather.location && weatherData.length == 0">
+	    <i class="fa fa-frown-o" aria-label="Error"></i>
+        <div class="error-message-weather-widget"> Oops! The weather service seems to be down. <a href="http://www.weather.com/weather/today/l/USWI0411:1:US" target="_blank">Try clicking here for local weather!</a></div>
```

Comment:
Maybe "Try clicking here for Madison weather." instead, since Madison might not be local from the perspective of whoever is viewing the widget?